### PR TITLE
Fix soft keyboard issue in backdrop.

### PR DESCRIPTION
**Pull request by akashmehta**
Fix following issue in all the projects onwards unit_7 : Key board remains open after selecting different item from back panel.

Resolve issue following issue in unit_converter project: front panel open twice on selecting hamburger icon.

### CODE REVIEW

**Comment by akashmehta** [comment]
> **Created new branch to resolve issue in cla/google occur by commit from wrong user name.**

Open issue: 
After opening front panel by selecting any category, open soft keyboard by clicking input field. After that if when user clicks on hamburger icon the front panel and soft keyboard closed and front panel opens again automatically.

Expected behaviour :
When user clicks on hamburger icon soft keyboard should be closed and front panel should not open automatically again.

This issue was created due to MediaQuery used in category_route.dart. I resolved that issue in [unit_converter app](https://github.com/flutter/udacity-course/tree/master/unit_converter/unit_converter) by using OrientationBuilder for responsive design.

Did not change in any other project because in video series MediaQuery is used to explain the concept of Responsive design.

@maryx Please review this pull request instead of [PR : 145](https://github.com/flutter/udacity-course/pull/145). I generated new pull request because cla/google reported an issue because of commit made by wrong user.

Thank you.

File: unit_converter/unit_converter/lib/category_route.dart [hunk 1]
```diff
@@ -221,7 +188,34 @@ class _CategoryRouteState extends State<CategoryRoute> {
         right: 8.0,
         bottom: 48.0,
       ),
-      child: _buildCategoryWidgets(MediaQuery.of(context).orientation),
+      child: OrientationBuilder(builder: (context, orientation) {
+        if (orientation == Orientation.portrait) {
```
Comment (maryx) [inline unit_converter/unit_converter/lib/category_route.dart]: Can you still put this into the `_buildCategoryWidgets()` helper function? It makes the code a little cleaner as there isn't as much nesting.

File: unit_converter/unit_converter/lib/category_route.dart [hunk 2]
```diff
@@ -214,14 +216,13 @@ class _CategoryRouteState extends State<CategoryRoute> {
...
-    assert(debugCheckHasMediaQuery(context));
-    final listView = Padding(
+    final listView = Padding (
       padding: EdgeInsets.only(
         left: 8.0,
         right: 8.0,
         bottom: 48.0,
       ),
-      child: _buildCategoryWidgets(MediaQuery.of(context).orientation),
+      child: _buildCategoryWidgets()
```
Comment (maryx) [inline unit_converter/unit_converter/lib/category_route.dart]: Nitpick: Please add a comma after the `)`

**Comment by maryx** [comment]
@akashmehta Re: your comment about the helper function being an antipattern/affecting performance, it's valid. However, this build() is quite small so there shouldn't be much of a performance hit. I also don't want this code to diverge too much from the code in the course exercises, or create a lot of widget classes, which may look intimidating.

File: unit_converter/unit_converter/lib/category_route.dart (hunk 2 above)
Comment (akashmehta) [reply unit_converter/unit_converter/lib/category_route.dart]: @maryx  change applied, Please recheck.

**Comment by akashmehta** [comment]
I agree that it will look to intimidating to use too many widget classes, so we found one third party library named [functional widget](https://github.com/rrousselGit/functional_widget) for that to reduce boiler plate code. Since our code is not that large so in this case we don't need to worry about it. :+1:

**Comment by maryx** [comment]
LGTM. Thanks for contributing and your quick responses!

**Pull request merged**